tests/test_builtin_settings.py

```diff
@@ -0,0 +1,393 @@
+        assert (
+            ReportPlot(solver=solver, new_instance_name="report-plot-1")
+            == solver.solution.monitor.report_plots["report-plot-1"]
+        )
+        assert (
+            ReportPlot(solver=solver, name="report-plot-1")
+            == solver.solution.monitor.report_plots["report-plot-1"]
+        )
+        if solver.get_fluent_version() >= FluentVersion.v251:
+            assert (
+                ReportPlot(solver=solver)
+                == solver.solution.monitor.report_plots["report-plot-2"]
+            )
+        assert (
+            ConvergenceConditions(solver=solver)
+            == solver.solution.monitor.convergence_conditions
+        )
+    else:
+        with pytest.raises(RuntimeError):
+            Monitor(solver=solver)
+    if solver.get_fluent_version() >= FluentVersion.v231:
+        assert CellRegisters(solver=solver) == solver.solution.cell_registers
+        assert (
+            CellRegister(solver=solver, new_instance_name="cell_register_1")
+            == solver.solution.cell_registers["cell_register_1"]
+        )
+        assert (
+            CellRegister(solver=solver, name="cell_register_1")
+            == solver.solution.cell_registers["cell_register_1"]
+        )
+        if solver.get_fluent_version() >= FluentVersion.v251:
+            assert (
+                CellRegister(solver=solver)
+                == solver.solution.cell_registers["cell_register_2"]
+            )
+    else:
+        with pytest.raises(RuntimeError):
+            CellRegisters(solver=solver)
+    assert Initialization(solver=solver) == solver.solution.initialization
+    if solver.get_fluent_version() >= FluentVersion.v231:
+        assert (
+            CalculationActivity(solver=solver) == solver.solution.calculation_activity
+        )
+        assert (
+            ExecuteCommands(solver=solver)
+            == solver.solution.calculation_activity.execute_commands
+        )
+        if solver.get_fluent_version() >= FluentVersion.v241:
+            assert (
+                CaseModification(solver=solver)
+                == solver.solution.calculation_activity.case_modification
+            )
+        else:
+            with pytest.raises(RuntimeError):
+                CaseModification(solver=solver)
+    else:
+        with pytest.raises(RuntimeError):
+            CalculationActivity(solver=solver)
+    assert RunCalculation(solver=solver) == solver.solution.run_calculation
+
+
+def test_builtin_singleton_setting_assign_session(
+    new_meshing_session, new_solver_session
+):
+    meshing = new_meshing_session
+    solver = new_solver_session
+
+    models = Models()
+    assert isinstance(models, Models)
+    with pytest.raises(TypeError):
+        models.solver = meshing
+    models.solver = solver
+    assert not isinstance(models, Models)
+    assert models.path == "setup/models"
+    assert not models.is_active()
```

Comment:
This doesn't hold in general. E.g., `models.viscous` is active, even though `models` is inactive. This needs to be fixed in the server-side.